### ISSUE TITLE
Error 500 when requesting exchanges with the management plugin

### ISSUE THREAD

**Issue by dereckson** [rabbitmq/rabbitmq-server#7813]
### Describe the bug

A 500 issue occurs when requesting exchanges with the management plugin, web or through a HTTP call.

Logs show a process crash:
```
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0> ** Generic server <0.1853.0> terminating
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0> ** Last message in was {submit,#Fun<rabbit_mgmt_db.21.88141441>,<0.1951.0>,
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>                                reuse}
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0> ** When Server state == {from,<0.1951.0>,#Ref<0.2940823630.563085314.233689>}
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0> ** Reason for termination ==
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0> ** {{badkey,{resource,<<"dev">>,exchange,<<"notifications">>}},
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>     [{erlang,map_get,
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>              [{resource,<<"dev">>,exchange,<<"notifications">>},#{}],
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>              [{error_info,#{module => erl_erts_errors}}]},
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>      {rabbit_mgmt_db,'-detail_exchange_stats/3-lc$^1/1-1-',4,
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>                      [{file,"rabbit_mgmt_db.erl"},{line,548}]},
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>      {worker_pool_worker,handle_call,3,
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>                          [{file,"worker_pool_worker.erl"},{line,96}]},
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>      {gen_server2,handle_msg,2,[{file,"gen_server2.erl"},{line,1046}]},
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>      {proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,250}]}]}
2023-04-03 23:28:30.626472+00:00 [error] <0.1853.0>
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>   crasher:
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>     initial call: worker_pool_worker:init/1
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>     pid: <0.1853.0>
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>     registered_name: []
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>     exception exit: {{badkey,
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                          {resource,<<"dev">>,exchange,<<"notifications">>}},
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                      [{erlang,map_get,
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                           [{resource,<<"dev">>,exchange,<<"notifications">>},
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                            #{}],
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                           [{error_info,#{module => erl_erts_errors}}]},
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                       {rabbit_mgmt_db,'-detail_exchange_stats/3-lc$^1/1-1-',
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                           4,
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                           [{file,"rabbit_mgmt_db.erl"},{line,548}]},
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                       {worker_pool_worker,handle_call,3,
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                           [{file,"worker_pool_worker.erl"},{line,96}]},
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                       {gen_server2,handle_msg,2,
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                           [{file,"gen_server2.erl"},{line,1046}]},
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                       {proc_lib,wake_up,3,
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>                           [{file,"proc_lib.erl"},{line,250}]}]}
2023-04-03 23:28:30.627210+00:00 [error] <0.1853.0>
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>   crasher:
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     initial call: cowboy_stream_h:request_process/3
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     pid: <0.1951.0>
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     registered_name: []
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     exception exit: {{{{badkey,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                            {resource,<<"dev">>,exchange,<<"notifications">>}},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                        [{erlang,map_get,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                             [{resource,<<"dev">>,exchange,<<"notifications">>},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                              #{}],
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                             [{error_info,#{module => erl_erts_errors}}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                         {rabbit_mgmt_db,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                             '-detail_exchange_stats/3-lc$^1/1-1-',4,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                             [{file,"rabbit_mgmt_db.erl"},{line,548}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                         {worker_pool_worker,handle_call,3,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                             [{file,"worker_pool_worker.erl"},{line,96}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                         {gen_server2,handle_msg,2,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                             [{file,"gen_server2.erl"},{line,1046}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                         {proc_lib,wake_up,3,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                             [{file,"proc_lib.erl"},{line,250}]}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {gen_server2,call,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [<0.1853.0>,                                                                                                                                2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                            {submit,#Fun<rabbit_mgmt_db.21.88141441>,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                                <0.1951.0>,reuse},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                            infinity]}},                                                                                                                               2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                      [{gen_server2,call,3,                                                                                                                            2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"gen_server2.erl"},{line,346}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {rabbit_mgmt_wm_exchange,to_json,2,                                                                                                             2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"rabbit_mgmt_wm_exchange.erl"},{line,44}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {cowboy_rest,call,3,                                                                                                                            2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"src/cowboy_rest.erl"},{line,1575}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {cowboy_rest,set_resp_body,2,                                                                                                                   2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"src/cowboy_rest.erl"},{line,1464}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {cowboy_rest,upgrade,4,                                                                                                                         2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"src/cowboy_rest.erl"},{line,284}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {cowboy_stream_h,execute,3,                                                                                                                     2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"src/cowboy_stream_h.erl"},{line,300}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {cowboy_stream_h,request_process,3,                                                                                                             2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"src/cowboy_stream_h.erl"},{line,291}]},
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                       {proc_lib,init_p_do_apply,3,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                           [{file,"proc_lib.erl"},{line,240}]}]}
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>       in function  gen_server2:call/3 (gen_server2.erl, line 346)
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>       in call from rabbit_mgmt_wm_exchange:to_json/2 (rabbit_mgmt_wm_exchange.erl, line 44)
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>       in call from cowboy_rest:call/3 (src/cowboy_rest.erl, line 1575)
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>       in call from cowboy_rest:set_resp_body/2 (src/cowboy_rest.erl, line 1464)
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>       in call from cowboy_rest:upgrade/4 (src/cowboy_rest.erl, line 284)
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>       in call from cowboy_stream_h:execute/3 (src/cowboy_stream_h.erl, line 300)
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>       in call from cowboy_stream_h:request_process/3 (src/cowboy_stream_h.erl, line 291)
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     ancestors: [<0.1950.0>,<0.765.0>,<0.759.0>,<0.758.0>,<0.756.0>,
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>                   rabbit_web_dispatch_sup,<0.745.0>]
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     message_queue_len: 0
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     messages: []
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     links: [<0.1950.0>]
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     dictionary: [{{xtype_to_module,topic},rabbit_exchange_type_topic}]
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     trap_exit: false
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     status: running
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     heap_size: 6772
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     stack_size: 28
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>     reductions: 2630
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>   neighbours:
2023-04-03 23:28:30.629194+00:00 [error] <0.1951.0>
2023-04-03 23:28:30.632404+00:00 [error] <0.1950.0> Ranch listener {acceptor,{0,0,0,0,0,0,0,0},15672}, connection process <0.1950.0>, stream 1 had its request process <0.1951.0> exit with reason {{{badkey,{resource,<<"
dev">>,exchange,<<"notifications">>}},[{erlang,map_get,[{resource,<<"dev">>,exchange,<<"notifications">>},#{}],[{error_info,#{module => erl_erts_errors}}]},{rabbit_mgmt_db,'-detail_exchange_stats/3-lc$^1/1-1-',4,[{file
,"rabbit_mgmt_db.erl"},{line,548}]},{worker_pool_worker,handle_call,3,[{file,"worker_pool_worker.erl"},{line,96}]},{gen_server2,handle_msg,2,[{file,"gen_server2.erl"},{line,1046}]},{proc_lib,wake_up,3,[{file,"proc_lib.
erl"},{line,250}]}]},{gen_server2,call,[<0.1853.0>,{submit,#Fun<rabbit_mgmt_db.21.88141441>,<0.1951.0>,reuse},infinity]}} and stacktrace [{gen_server2,call,3,[{file,"gen_server2.erl"},{line,346}]},{rabbit_mgmt_wm_excha
nge,to_json,2,[{file,"rabbit_mgmt_wm_exchange.erl"},{line,44}]},{cowboy_rest,call,3,[{file,"src/cowboy_rest.erl"},{line,1575}]},{cowboy_rest,set_resp_body,2,[{file,"src/cowboy_rest.erl"},{line,1464}]},{cowboy_rest,upgr
ade,4,[{file,"src/cowboy_rest.erl"},{line,284}]},{cowboy_stream_h,execute,3,[{file,"src/cowboy_stream_h.erl"},{line,300}]},{cowboy_stream_h,request_process,3,[{file,"src/cowboy_stream_h.erl"},{line,291}]},{proc_lib,ini
t_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]
2023-04-03 23:28:30.632404+00:00 [error] <0.1950.0>
```

### Reproduction steps

1. Browse UI for exchanges, or send a GET request to /api/exchanges/<vhost>/<exchange>
2. Get a 500

### Expected behavior

A error message explaining what happens.

### Additional context

RabbitMQ 3.11.8 + Erlang:      25.2.2 [jit]

Standard configuration, run in Docker

Another occurence of this stacktrace: https://ask.csdn.net/questions/7791019

Users, users permissions, queues or queues bindings work fine.

**Comment by dereckson** [comment]
I can repro when deleting container and storage, and redeploying.
I CANNOT repro when upgrading to 3.11.11

**Comment by michaelklishin** [comment]
#7369, #7301, and a couple of similar issues were addressed after 3.11.8.

Not sure what else you expect us to do if you cannot reproduce with 3.11.11.

The latest RabbitMQ release is 3.11.13.
